### PR TITLE
Add local registry initialization to Makefile

### LINKED ISSUE #25: Prefer local development use cases for environment variable defaults
**Is your feature request related to a problem? Please describe.**
As an engineer trying to get started with this build tooling, I must first configure several environment variables.

**Describe the solution you'd like**
***Preferred Experience***
Copy the Makefile into a project and run `make init` without having to configure anything else.

***High Level Design***
Today the init process downloads a container and an image. If the init process also [deployed a local registry server](https://docs.docker.com/registry/deploying/) then there would be no need to require the user to configure additional values to point to a docker registry, or provide credentials.

They'd still be free to change these values if they have more specialized needs, but this would provide a configuration that worked out of the box.

A single local registry could be shared between multiple projects. This might lead to a design decision for build-tooling to maintain some stateful data on a local development machine. The stateful information relevant to this feature would be the endpoint (including port) to access the local registry. This would allow other projects to check if a local registry has already been deployed and started.

**Additional context**
By choosing this pattern we are optimizing the local development experience. This isn't really at the expense of the CI pipeline experience, because there is no reliable way to optimize defaults for CI pipelines across all intended users. Those workflows will require environment variables to be overridden. That is a reasonable expectation of a pipeline configuration.

CC: @navidshaikh

### PR DESCRIPTION
**Pull request by jmoroski**
# Description
This change adds a make target to create a local Docker registry for pushing and pulling packages without requiring additional configuration or external registry access.

This change also introduces a bootstrap target that aggregates the environment and project initialization process across a few finer-grained targets. The behavior of this new target is as follows:
1. Start a docker registry container locally (if necessary) a. This can be bypassed in a CI environment
2. Check for the presence of a build container Dockerfile in working dir a. Prompt to overwrite if found b. Add build container Dockerfile if not present or if overwriting
3. Pull the Tanzu packaging image

Other minor changes:
* Changed how help was being generated to support multiline text and classification of targets
* Made help the default target

Fixes #25

## Change Details
Check all that apply:
- [x] New feature 
- [ ] Bug fix 
- [ ] Non-code change 
- [x] Non-breaking change 
- [ ] Breaking change
- [x] Requires a documentation change

## Release Note

```release-note
Adds a local registry via the bootstrap target to simplify local development flows.
```

# How Has This Been Tested?
1. Ran the make bootstrap target in a new project. Verified that a local registry container was started, a Dockerfile and lint configuration were added to the root diretory of the project, and the package image was pulled down
2. Ran make boostrap again. Verified that the registry container was identified to be already running. Verified that I was prompted to overwrite an existing Dockerfile. Selected 'N' and verified that the Dockerfile was unchanged.
3. Ran make bootstrap again. Verified that I was prompted to overwrite existing Dockerfile. Selected 'Y' and verified that the Dockerfile was overwritten to the original state.
4. Ran both make and make help. Verified that the help message was output in both cases.

# Checklist:
- [x] My code follows the [Contribution and Style guidelines](../../CONTRIBUTING.md) of this project
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] I have added test details that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by yharish991 — CHANGES_REQUESTED** [review]

File: templates/Makefile [hunk 1]
```diff
@@ -6,16 +16,69 @@ BUILD_TOOLING_CONTAINER_IMAGE ?= ghcr.io/vmware-tanzu/build-tooling
 PACKAGING_CONTAINER_IMAGE ?= ghcr.io/vmware-tanzu/package-tooling
 VERSION ?= v0.0.1
 
+##
+## Project Targets
```
[the remark this answers is not in the document]
Comment (jmoroski) [reply templates/Makefile:20]: Absolutely. I'll make that change

File: templates/Makefile [hunk 2]
```diff
@@ -1,3 +1,13 @@
+.DEFAULT_GOAL := help
+
+REGISTRY_PORT := 5000
```
[the remark this answers is not in the document]
Comment (jmoroski) [reply templates/Makefile:3]: Good to know. I'll find something that works as a better universal default.

[the remark this answers is not in the document]
Comment (jmoroski) [reply templates/Makefile]: Updated to use port 8000 which seems to align with where docker is headed to resolve this issue (https://github.com/docker/docs/issues/14226)

File: templates/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmoroski) [reply templates/Makefile]: changed

**Review by yharish991 — APPROVED** [review]

**Pull request merged**